src/pybind/mgr/dashboard/frontend/src/app/shared/api/performance-counter.service.spec.ts

```diff
@@ -1,22 +1,39 @@
-import { HttpClientModule } from '@angular/common/http';
-import { HttpClientTestingModule } from '@angular/common/http/testing';
-import { inject } from '@angular/core/testing';
-
-import { BsDropdownModule } from 'ngx-bootstrap';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
 
 import { configureTestBed } from '../unit-test-helper';
 import { PerformanceCounterService } from './performance-counter.service';
 
 describe('PerformanceCounterService', () => {
+  let service: PerformanceCounterService;
+  let httpTesting: HttpTestingController;
+
   configureTestBed({
     providers: [PerformanceCounterService],
-    imports: [HttpClientTestingModule, BsDropdownModule.forRoot(), HttpClientModule]
+    imports: [HttpClientTestingModule]
   });
 
-  it(
-    'should be created',
-    inject([PerformanceCounterService], (service: PerformanceCounterService) => {
-      expect(service).toBeTruthy();
-    })
-  );
+  beforeEach(() => {
+    service = TestBed.get(PerformanceCounterService);
+    httpTesting = TestBed.get(HttpTestingController);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
+
+  it('should call list', () => {
+    service.list().subscribe();
+    const req = httpTesting.expectOne('api/perf_counters');
+    expect(req.request.method).toBe('GET');
+    httpTesting.verify();
+  });
+
+  it('should call get', () => {
+    service.get('foo', '1').subscribe();
+    const req = httpTesting.expectOne('api/perf_counters/foo/1');
+    req.flush([{ counters: {} }]);
```

Comment:
This line can be removed as the outcome is not used.

You should check all your flushes if they are in use or not. If they should be used implement them in a way they are used else please remove them

Reply:
I'm sure that the subscribe resolves, but the result is not tested. It could be possible that some service will change over time and somehow change the result in a unsuspected way, this is why you should test the result of the observable that is returned by the function you test.

Reply:
I am now verifying the result on tests that are using `flush`.